compiler/rustc_ast/src/ast.rs

```diff
@@ -2485,6 +2485,13 @@ pub enum Unsafe {
     No,
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Encodable, Decodable, Debug)]
+#[derive(HashStable_Generic)]
+pub enum FnSafety {
+    Unsafe(Span),
+    Default,
```

Comment:
Maybe name this `Inherited` or `FromContext`. Without an explanation it's not obvious what this is supposed to be

Reply:
Yeah, I've used Default as a way of saying no safety explicitly written means is going to take a default value.
Another possible name could be `Implicit`.
Out of all the values mentioned I slightly prefer `Default` but I don't mind changing it neither.
Let's see if somebody else have some thought about this.